sql/src/main/java/io/crate/executor/transport/RerouteActions.java

```diff
@@ -73,6 +73,17 @@ private RerouteActions() {
         return listener;
     }
 
+    public static CompletableFuture<Long> executeRetryFailed(
+        BiConsumer<ClusterRerouteRequest, ActionListener<ClusterRerouteResponse>> rerouteAction) {
+
+        ClusterRerouteRequest request = new ClusterRerouteRequest();
+        request.setRetryFailed(true);
+        FutureActionListener<ClusterRerouteResponse, Long> listener =
+            new FutureActionListener<>(r -> r.isAcknowledged() ? (long) r.getState().getRoutingNodes().unassigned().size() : -1L);
```

Comment:
Is this safe to assume that all unassigned shards are now being processed? (Doesn't the response contain the *updated* state where the shards should now be assigned?)